firestore/src/common/filter.cc

```diff
@@ -0,0 +1,172 @@
+  return GreaterThan(FieldPath::FromDotSeparatedString(field), value);
+}
+
+Filter Filter::GreaterThanOrEqualTo(const std::string& field,
+                                    const FieldValue& value) {
+  return GreaterThanOrEqualTo(FieldPath::FromDotSeparatedString(field), value);
+}
+
+Filter Filter::LessThan(const std::string& field, const FieldValue& value) {
+  return LessThan(FieldPath::FromDotSeparatedString(field), value);
+}
+
+Filter Filter::LessThanOrEqualTo(const std::string& field,
+                                 const FieldValue& value) {
+  return LessThanOrEqualTo(FieldPath::FromDotSeparatedString(field), value);
+}
+
+Filter Filter::In(const std::string& field,
+                  const std::vector<FieldValue>& values) {
+  return In(FieldPath::FromDotSeparatedString(field), values);
+}
+
+Filter Filter::NotIn(const std::string& field,
+                     const std::vector<FieldValue>& values) {
+  return NotIn(FieldPath::FromDotSeparatedString(field), values);
+}
+
+Filter Filter::ArrayContains(const FieldPath& field, const FieldValue& value) {
+  return FilterInternal::ArrayContains(field, value);
+}
+
+Filter Filter::ArrayContainsAny(const FieldPath& field,
+                                const std::vector<FieldValue>& values) {
+  return FilterInternal::ArrayContainsAny(field, values);
+}
+
+Filter Filter::EqualTo(const FieldPath& field, const FieldValue& value) {
+  return FilterInternal::EqualTo(field, value);
+}
+
+Filter Filter::NotEqualTo(const FieldPath& field, const FieldValue& value) {
+  return FilterInternal::NotEqualTo(field, value);
+}
+
+Filter Filter::GreaterThan(const FieldPath& field, const FieldValue& value) {
+  return FilterInternal::GreaterThan(field, value);
+}
+
+Filter Filter::GreaterThanOrEqualTo(const FieldPath& field,
+                                    const FieldValue& value) {
+  return FilterInternal::GreaterThanOrEqualTo(field, value);
+}
+
+Filter Filter::LessThan(const FieldPath& field, const FieldValue& value) {
+  return FilterInternal::LessThan(field, value);
+}
+
+Filter Filter::LessThanOrEqualTo(const FieldPath& field,
+                                 const FieldValue& value) {
+  return FilterInternal::LessThanOrEqualTo(field, value);
+}
+
+Filter Filter::In(const FieldPath& field,
+                  const std::vector<FieldValue>& values) {
+  return FilterInternal::In(field, values);
+}
+
+Filter Filter::NotIn(const FieldPath& field,
+                     const std::vector<FieldValue>& values) {
+  return FilterInternal::NotIn(field, values);
+}
+
+template <typename... Filters>
+Filter Filter::Or(const Filter& filter, const Filters&... filters) {
+  return FilterInternal::Or(filter, filters...);
+}
+
+template <typename... Filters>
+Filter Filter::And(const Filter& filter, const Filters&... filters) {
+  return FilterInternal::And(filter, filters...);
+}
+
+bool operator==(const Filter& lhs, const Filter& rhs) {
+  return &lhs == &rhs || *lhs.internal_ == *rhs.internal_;
```

Comment:
I think you should use the helper function that was created and used [elsewhere](https://cs.opensource.google/firebase-sdk/firebase-cpp-sdk/+/main:firestore/src/common/field_value.cc;l=344-346?q=field_value.cc&ss=firebase-sdk%2Ffirebase-cpp-sdk)

```cpp
return EqualityCompare(lhs.internal_, rhs.internal_);
```